nestjs prod build fail due to externalDependencies

_[Please make sure you have read the submission guidelines before posting an issue](https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md#-submitting-issue)_

# Prerequisites

Please answer the following questions for yourself before submitting an issue.
**YOU MAY DELETE THE PREREQUISITES SECTION.**

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed
- [x] I'm reporting the issue to the correct repository (not related to Angular, AngularCLI or any dependency)

## Expected Behavior
sample project : https://github.com/xmlking/ngx-starter-kit

`ng build api --prod` should work without warnings and errors

## Current Behavior

`ng build api ` works but   `ng build api --prod` throw following error

> PS: if I remove `"externalDependencies": "none"` from `angular.json` `api` project, it works
 
```
WARNING in ./node_modules/ws/lib/validation.js
Module not found: Error: Can't resolve 'utf-8-validate' in '/Developer/Work/SPA/ngx-starter-kit/node_modules/ws/lib'
 @ ./node_modules/ws/lib/validation.js
 @ ./node_modules/ws/lib/sender.js
 @ ./node_modules/ws/index.js
 @ ./node_modules/engine.io/lib/server.js
 @ ./node_modules/engine.io/lib/engine.io.js
 @ ./node_modules/socket.io/lib/index.js
 @ ./apps/api/src/app/shared/eventbus.gateway.ts
 @ ./apps/api/src/app/shared/index.ts
 @ ./apps/api/src/app/user/user.module.ts
 @ ./apps/api/src/app/user/index.ts
 @ ./apps/api/src/app/app.module.ts
 @ ./apps/api/src/main.ts
 @ multi ./apps/api/src/main.ts

ERROR in ./node_modules/got/index.js
Module not found: Error: Can't resolve 'electron' in '/Developer/Work/SPA/ngx-starter-kit/node_modules/got'
 @ ./node_modules/got/index.js 75:29-48
 @ ./node_modules/openid-client/lib/helpers/http.js
 @ ./node_modules/openid-client/lib/issuer.js
 @ ./node_modules/openid-client/lib/index.js
 @ ./node_modules/kubernetes-client/lib/auth-providers/openid.js
 @ ./node_modules/kubernetes-client/lib/auth-providers sync ^\.\/.*\.js$
 @ ./node_modules/kubernetes-client/lib/backends/request.js
 @ ./node_modules/kubernetes-client/lib/swagger-client.js
 @ ./node_modules/kubernetes-client/lib/index.js
 @ ./apps/api/src/app/project/kubernetes/kubernetes.service.ts
 @ ./apps/api/src/app/project/project.module.ts
 @ ./apps/api/src/app/project/index.ts
 @ ./apps/api/src/app/app.module.ts
 @ ./apps/api/src/main.ts
 @ multi ./apps/api/src/main.ts

ERROR in ./node_modules/pg/lib/native/client.js
Module not found: Error: Can't resolve 'pg-native' in '/Developer/Work/SPA/ngx-starter-kit/node_modules/pg/lib/native'
 @ ./node_modules/pg/lib/native/client.js 9:13-33
 @ ./node_modules/pg/lib/native/index.js
 @ ./node_modules/pg/lib/index.js
 @ ./node_modules/typeorm/platform/PlatformTools.js
 @ ./node_modules/typeorm/entity-manager/MongoEntityManager.js
 @ ./node_modules/typeorm/index.js
 @ ./apps/api/src/app/auth/auth.service.ts
 @ ./apps/api/src/app/auth/index.ts
 @ ./apps/api/src/app/app.module.ts
 @ ./apps/api/src/main.ts
 @ multi ./apps/api/src/main.ts

ERROR in ./node_modules/engine.io/lib/server.js
Module not found: Error: Can't resolve 'uws' in '/Developer/Work/SPA/ngx-starter-kit/node_modules/engine.io/lib'
 @ ./node_modules/engine.io/lib/server.js 90:23-37
 @ ./node_modules/engine.io/lib/engine.io.js
 @ ./node_modules/socket.io/lib/index.js
 @ ./apps/api/src/app/shared/eventbus.gateway.ts
 @ ./apps/api/src/app/shared/index.ts
 @ ./apps/api/src/app/user/user.module.ts
 @ ./apps/api/src/app/user/index.ts
 @ ./apps/api/src/app/app.module.ts
 @ ./apps/api/src/main.ts
 @ multi ./apps/api/src/main.ts

ERROR in ./node_modules/sharp/build/Release/sharp.node 1:0
Module parse failed: Unexpected character '�' (1:0)
You may need an appropriate loader to handle this file type.
(Source code omitted for this binary file)
 @ ./node_modules/sharp/lib/utility.js 3:14-52
 @ ./node_modules/sharp/lib/index.js
 @ ./apps/api/src/app/user/profile/profile.service.ts
 @ ./apps/api/src/app/user/user.module.ts
 @ ./apps/api/src/app/user/index.ts
 @ ./apps/api/src/app/app.module.ts
 @ ./apps/api/src/main.ts
 @ multi ./apps/api/src/main.ts
```
 

 

### Context

sample project : https://github.com/xmlking/ngx-starter-kit

Please provide any relevant information about your setup:

- version of Nx used
- version of Angular CLI used
- `angular.json` configuration
- version of Angular DevKit used
- 3rd-party libraries and their versions
- and most importantly - a use-case that fails

**A minimal reproduce scenario using allows us to quickly confirm a bug (or point out coding problem) as well as confirm that we are fixing the right problem.**

### Failure Logs

Please include any relevant log snippets or files here.

## Other

Any other relevant information that will help us help you.

--- a/packages/schematics/src/collection/node-application/index.ts
+++ b/packages/schematics/src/collection/node-application/index.ts
@@ -126,8 +126,7 @@ function getBuildConfig(project: any, options: NormalizedSchema) {
             replace: join(project.sourceRoot, 'environments/environment.ts'),
             with: join(project.sourceRoot, 'environments/environment.prod.ts')
           }
-        ],
-        externalDependencies: 'none'
+        ]
       }
     }
   };

--- a/packages/schematics/src/collection/node-application/node-application.spec.ts
+++ b/packages/schematics/src/collection/node-application/node-application.spec.ts
@@ -48,8 +48,7 @@ describe('node-app', () => {
                     with:
                       'apps/my-node-app/src/environments/environment.prod.ts'
                   }
-                ],
-                externalDependencies: 'none'
+                ]
               }
             }
           },

